core/src/main/scala/org/typelevel/vault/Vault.scala

```diff
@@ -45,6 +45,11 @@ final class Vault private (private val m: Map[Unique.Token, Locker]) {
    */
   private[vault] def lookup[A](k: Key[A]): Option[A] = lookup(k: LookupKey[A])
 
+  /**
+   * Checks if the value of a key is in this vault
+   */
+  def contains[A](k: LookupKey[A]): Boolean = m.contains(k.unique)
```

Comment:
Hm, I don't think this needs a type parameter. I realize it's poorly named now, but it can just ask for a `DeleteKey`. Or a `LookupKey[Any]` could work too, but would prevent running `contains` with an `InsertKey`. Maybe that's the right semantic though? 🤔

Reply:
Oh interesting. I did not consider the design space of the various key types. I'll look into this a bit more shortly.

Reply:
I don't see any reason to reject an `InsertKey`, other than it turns a nice `lookup(k).isDefined <=> contains(k)` property into "... if it compiles".

Reply:
I realize I haven't used Vault enough to comment on the ideal API here, so I'll defer to you two.

Taking the exact same input as `lookup` was just the easy thing to do.

Reply:
We start with `Key`, and whether we `.map` or `.contramap` it to something else, it shouldn't affect the `contains` relationship.  I'm in favor of `DeleteKey`, with an explanation of the quirky name in the scaladoc.

Reply:
FWIW: Ross's explanation makes sense to me.

Reply:
I didn't quite explain the name, I just mentioned it could accept any type of key, how's that?
```
   * Checks if the value of a key is in this vault.
   *
   * @param k
   *   the key to lookup can be any LookupKey, InsertKey, or DeleteKey
   *
```